### PR TITLE
Tsc 520 changes to account menus

### PR DESCRIPTION
**Pull request by sbal15**
pic1
image

removed dark/light mode & language from profile
added preferences tab

### CODE REVIEW

**Comment by paologum** [comment]
because language and dark mode are not directly correlated, i would probably move towards a hover menu button labeled `Language` that opens a second dropdown to the side that has the languages

**Comment by sbal15** [comment]
image

**Comment by sbal15** [comment]
image

File: app/src/NavBar.tsx [hunk 1]
```diff
@@ -179,6 +160,70 @@ export const NavBar = observer(function Navbar() {
             }}
           />
         )}
+
+        <PopupState variant="popover" popupId="demo-popup-menu">
+          {(popupState) => (
+            <React.Fragment>
+              <Button variant="text" {...bindTrigger(popupState)}>
+                <SettingsIcon />
+                <Typography>{t(``)}</Typography>
```
Comment (paologum) [inline app/src/NavBar.tsx:169]: if there is not translation just remove the `Typography`

File: app/src/NavBar.tsx [hunk 2]
```diff
@@ -179,6 +160,70 @@ export const NavBar = observer(function Navbar() {
...
+              </Button>
+              <Menu {...bindMenu(popupState)}>
+                <MenuItem className="settings-sub-menu-item" onClick={() => {}}>
+                  <Typography>{t(``)}</Typography>
```
Comment (paologum) [inline app/src/NavBar.tsx:173]: same here, also the above `onClick is empty so remove it

File: app/src/NavBar.tsx [hunk 3]
```diff
@@ -179,6 +160,70 @@ export const NavBar = observer(function Navbar() {
             }}
           />
         )}
+
+        <PopupState variant="popover" popupId="demo-popup-menu">
```
Comment (paologum) [inline app/src/NavBar.tsx:164]: remove the `demo-popup-menu`

File: app/src/NavBar.tsx [hunk 4]
```diff
@@ -179,6 +160,70 @@ export const NavBar = observer(function Navbar() {
...
+                />
+                <MenuItem>
+                  <Typography sx={{ textAlign: "center", width: "100%" }}>{t(`LANGUAGES`)}</Typography>
+                </MenuItem>
+                <Divider
+                  sx={{
+                    borderBottomWidth: 0.75,
+                    borderColor: "gray",
+                    my: 1
+                  }}
```
Comment (paologum) [inline app/src/NavBar.tsx:203]: move the `sx` to the `css` file

**Review by paologum — CHANGES_REQUESTED** [review]

File: app/src/NavBar.tsx (hunk 2 above)
Comment (sbal15) [reply app/src/NavBar.tsx:173]: when i try to remove the onclick, it throws an error, would you know why?

Comment (paologum) [reply app/src/NavBar.tsx:173]: does this menu item have any use if it is empty with an empty onClick function? what is the use of this

File: app/src/NavBar.tsx [hunk 5]
```diff
@@ -179,6 +160,55 @@ export const NavBar = observer(function Navbar() {
...
+                  />
+                </MenuItem>
+                <Divider className="div-bar-top" />
+                <MenuItem>
+                  <Typography>{t(`LANGUAGES`)}</Typography>
+                </MenuItem>
+                <Divider className="div-bar-bottom" />
+                {Object.entries(languageList).map(([key, value]) => (
+                  <MenuItem
+                    ref={dropRef}
```
Comment (paologum) [inline app/src/NavBar.tsx:192]: does this ref have a function? it just is being declared but never used

**Review by paologum — CHANGES_REQUESTED** [review]

File: app/src/NavBar.tsx (hunk 5 above)
Comment (sbal15) [reply app/src/NavBar.tsx:192]: oops sorry, forgot to take it out, i was experimenting with the drop down menu

**Comment by paologum** [comment]
tab selection seems to be buggy on the pr preview, there is probably a bug in this branch related to this because on local and dev sites this is not the case

https://github.com/user-attachments/assets/00f5389d-aa3e-4c0f-ac13-d7b6cfbab098

File: scripts/json-to-excel.ts
Comment (paologum) [inline scripts/json-to-excel.ts:1]: don't commit this

File: app/src/NavBar.tsx [hunk 6]
```diff
@@ -110,19 +88,19 @@ export const NavBar = observer(function Navbar() {
...
+              <Tab
+                value={3}
+                disableRipple
+                label={t("navBar.settings")}
+                to="/settings"
+                component={Link}
+                ref={settingsRef}
+                {...anchorProps}
+              />
+              <Tab value={3} disableRipple label={t("navBar.help")} to="/help" component={Link} />
```
Comment (paologum) [inline app/src/NavBar.tsx:102]: same tab value makes selection of tabs off

**Review by paologum — CHANGES_REQUESTED** [review]
looks like you removed the mobile functionality from the pr, you will need to add that back

**Review by paologum — APPROVED** [review]

**Pull request merged**